### PR TITLE
admin: add histogram rendering for html mode.

### PR DESCRIPTION
**Pull request by jmarantz**
Commit Message: Renders histograms in the stats HTML view. Screenshot:
Screenshot 2023-05-29 at 11 39 15 PM

This is the next phase in https://github.com/envoyproxy/envoy/pull/26472 -- remaining phases are further refactors, as well as sorting histograms by change-count for active-html mode, and make web navigation work better.

This includes unit tests for the new histogram rendering, as well as a mechanism to render histograms.js on its own.

Additional Description:
Risk Level: low
Testing: //test/...,  test/integration/admin_html/web_test.sh for browser-based tests on FF, and manually ran on Chrome as well.
Docs Changes: n/a -- the histograms were previously rendered as text with format=html; now they are rendered graphically, but detail about this is not in the docs currently.
Release Notes:
Platform Specific Features:

### CODE REVIEW

File: source/server/admin/html/histograms.js [hunk 1]
```diff
@@ -0,0 +1,628 @@
...
+    bucketSpan.style.left = this.leftPercent;
+
+    let showingCount = false;
+    if (++this.textIntervalIndex == this.textInterval) {
+      showingCount = true;
+      this.textIntervalIndex = 0;
+      this.drawBucketLabels(bucket, heightPercent);
+    }
+
+    bucketSpan.addEventListener('mouseover', showPopupFn(
```
[the remark this answers is not in the document]
Comment (jmarantz) [reply source/server/admin/html/histograms.js]: done & thanks! actually the popup does have child elements.

**Review by ravenblackx — APPROVED** [review]

**Comment by jmarantz** [comment]
Greg -- can you do a senior maintainer pass?

I'm glad i removed the interpolation -- I decided to go a different way in the UI layer after seeing how hard it is to get more than a few hundred buckets. I just draw text every N buckets once there are too many, and rely on popups to display the textual detail of whatever bucket you are hovering over.

**Review by ggreenway — APPROVED** [review]
LGTM, but I didn't look at js, css, html

**Pull request merged**